lib/ansible/modules/system/gconftool2.py

```diff
@@ -36,146 +38,233 @@
-                                                                     call_type,
-                                                                     self.key,
-                                                                     self.value)
-            elif call_type == 'unset':
-                cmd += "--unset {0}".format(self.key)
-
-            # Start external command
-            process = Popen([cmd], stdout=PIPE, stderr=PIPE, shell=True)
-
-            # In either case, we will capture the output
-            out = process.stdout.read()
-            err = process.stderr.read()
-
-            if len(err) > 0:
-                if fail_onerr:
-                    self.ansible.fail_json(msg='gconftool-2 failed with '
-                                               'error: %s' % (str(err)))
-            else:
-                changed = True
-
-        except OSError:
-            exception = get_exception()
-            self.ansible.fail_json(msg='gconftool-2 failed with exception: '
-                                       '%s' % exception)
-        return changed, out.rstrip()
+from ansible.module_utils.basic import AnsibleModule
+
+
+class GConfTool2(object):
+    """
+    Small wrapper class around the gconftool-2 binary that can be used for
+    manipulating a single preference in GConf repository.
+    """
+
+    def __init__(self, module, direct=False, config_source=None, check_mode=False):
+        """
+        Initialises instance of the class.
+
+        :param module: Ansible module instance used to signal failures and run commands.
+        :type module: AnsibleModule
+
+        :param direct: Specify whether the GConf repository should be accessed directly or through gconfd-2 daemon.
+                       If enabled, valid config_source must be passed in as well.
+        :type direct: bool
+
+        :param config_source: Configuratrion source to use in case direct access has been requested.
+        :type config_source: str
+
+        :param check_mode: Specify whether to only check if a change should be made or if to actually make a change.
+        :type check_mode: bool
+        """
+
+        self.module = module
+        self.base_command = ["gconftool-2"]
+        self.check_mode = check_mode
+
+        if direct:
+            self.base_command += ["--direct", "--config-source", config_source]
+
+    def get(self, key):
+        """
+        Retrieves current value associated with the key.
+
+        If an error occurs, a call will be made to AnsibleModule.fail_json.
+
+        :returns: string -- Value assigned to the provided key. If the key does not exist (i.e. no value is assigned), returns None.
+        """
+
+        # Set-up command to run.
+        command = list(self.base_command)
+        command += ["--get", key]
+
+        # Run the command and fetch standard return code, stdout, and stderr.
+        rc, out, err = self.module.run_command(command)
```

Comment:
Could try running the command with the env setup to use LANG=C 

You can specify the env for the command either with the 'environ_update' arg to run_command() or
by setting module.run_command_environ_update before the run command.